src/content/guides/creating-your-own-admission-controller/index.md

```diff
@@ -0,0 +1,219 @@
+
+Let's start creating an HTTP server with the certs mounted from the secret. The server will listen to the path `validate` as we defined in the webhook.
+
+__Note:__ The code examples have been stripped out to make it easier to understand. For a more detailed look, browse the [repository](github.com/gianstwarm/grumpy).
+
+```go
+
+  // Read the certs from the convined path and convert it to a X509 keypair
+  flag.StringVar(&tlscert, "tlsCertFile", "/etc/certs/cert.pem", "x509 Certificate for HTTPS.")
+	flag.StringVar(&tlskey, "tlsKeyFile", "/etc/certs/key.pem", "x509 private key to --tlsCertFile.")
+  certs, _ := tls.LoadX509KeyPair(tlscert, tlskey)
+
+  // Create a secure http server
+	server := &http.Server{
+		Addr:      ":8080",
+		TLSConfig: &tls.Config{Certificates: []tls.Certificate{certs}},
+	}
+
+	// Create a handler listening to the 'validate' path and start the server
+	gs := GrumpyServerHandler{}
+	mux.HandleFunc("/validate", gs.serve)
+	server.ListenAndServeTLS("", "")
+
+```
+
+Inside the grumpy package, we define a `serve` function which reads the request body, then it converts the data to a `Pod` data type and finally checks if the resource name is valid.
+
+```go
+	// Convert raw data in a Pod data type
+  raw := arRequest.Request.Object.Raw
+	pod := v1.Pod{}
+	json.Unmarshal(raw, &pod)
+
+  // Actual validation logic
+	if pod.Name != "smooth-app" {
+		return
+	}
+```
+
+In case the request name is not the expected one (`smooth-app`), our handler creates a response notifying of the rejection. Otherwise, it returns and Kubernetes API server will follow the processing the request.
+
+```go
+  // Create a response to return to the Kubernetes API
+  ar := v1beta1.AdmissionReview{
+		Response: &v1beta1.AdmissionResponse{
+			Allowed: false,
+			Result: &metav1.Status{
+				Message: "Keep calm and not add more crap in the cluster!",
+			},
+		},
+	}
+	resp, err := json.Marshal(ar)
+```
+
+# Conclusion
+
+As you can see from this tutorial, it is quite easy to implement a simple admission controller. Obviously, there is plenty of possibilities to make your cluster secure and harden it (accept known registries, forbid latest tags, ...). 
+
+At the same time, it holds great power, since it can influence the key components running in the cluster. As an example, you could block the CNI plugin from running in case you commit an error which may lead to the entire cluster being borked. So be careful and try to scope the admission logic to a namespace or a minor set of actions.
+
+Also, it is good to mention there are already some projects which leverage this pattern to enable high order functionality. As an example [gatekeeper](https://github.com/replicatedhq/gatekeeper/) uses admission webhooks to implement a policy engine ([OPA](https://www.openpolicyagent.org/)) to enforce policies over Cloud Native environments.
```

Comment:
instead of gatekeeper I would mention the kubernetes-policy-controller from Azure as that has been upstreamed into OPA now as official k8s controller for OPA: https://github.com/open-policy-agent/kubernetes-policy-controller

```suggestion
Also, it is good to mention there are already some projects which leverage this pattern to enable high order functionality. As an example [kubernetes-policy-controller](https://github.com/open-policy-agent/kubernetes-policy-controller) uses admission webhooks to implement a policy engine ([OPA](https://www.openpolicyagent.org/)) to enforce policies over Cloud Native environments.
```